### ISSUE TITLE
load/storeChunk: slice view

### ISSUE THREAD

**Issue by ax3l** [openPMD/openPMD-api#138]
The `A.storeChunk(data, offset, extent)` and `A.loadChunk(data, chunk_offset, chunk_extent)` methods should for syntactic sugar provide a slice-view access of the form:

```C++
// write (copy)
A[view] = data;

// returns: std::unique_ptr< A.dtype >
auto B = A[view];
// hm, maybe that's not possible, so let's say something like
auto B = A[view].astype< some_dtype >();
```

**Comment by ax3l** [comment]
Hm, maybe we can also implements views when we add the `xtensor` frontend

https://xtensor.readthedocs.io/en/latest/basic_usage.html

related to #32

**Comment by ax3l** [comment]
Hm, I think we will need this C++ `Slice` class even earlier if we want to have a seamless python API.
And two helpers for:
`Extent`, `Offset` -> `Slice` and `Slice` -> `Extent`, `Offset`

No need to expose it in the C++ frontend at this point, tough.

**Comment by ax3l** [comment]
Implemented with #458
